src/main/kotlin/com/openlattice/search/requests/EntityNeighborsFilterBulk.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.search.requests
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.openlattice.client.serialization.SerializationConstants
+import java.util.UUID
+import java.util.Optional
+
+data class EntityNeighborsFilterBulk(
+        @JsonProperty(SerializationConstants.ENTITY_KEY_IDS) val entityKeyIds: Map<UUID, Set<UUID>>,
+        @JsonProperty(SerializationConstants.SRC) val srcEntitySetIds: Optional<Set<UUID>>,
+        @JsonProperty(SerializationConstants.DST) val dstEntitySetIds: Optional<Set<UUID>>,
+        @JsonProperty(SerializationConstants.EDGE) val associationEntitySetIds: Optional<Set<UUID>>
+) {
+    constructor(
+            entityKeyIds: Map<UUID, Set<UUID>>
+    ) : this(entityKeyIds, Optional.empty(), Optional.empty(), Optional.empty())
+
+    constructor(
+            entitySetId: UUID,
+            filter: EntityNeighborsFilter
+    ) : this(
+            mapOf(entitySetId to filter.entityKeyIds),
+            filter.srcEntitySetIds,
+            filter.dstEntitySetIds,
+            filter.associationEntitySetIds
+    )
+
+    init {
+        require(entityKeyIds.isNotEmpty()) { "EntityKeyIds cannot be empty in filter." }
```

Comment:
should these `require`s be `check`s?

Reply:
Both have the same implementation and serve the same purpose, but as far as I know, we rather use `require` to check function arguments and `check` to check objects state.

Reply:
aren't you checking the state of this object though? (the EntityNeighborsFilterBulk object)

Reply:
`require` throws an `IllegalArgumentException` and check an `IllegalStateException` and in this case, we check the arguments of the `EntityNeighborsFilterBulk` constructor, which I think should throw rather an `IllegalArgumentException` if any of the checks fail.
```
IllegalStateException
Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

IllegalArgumentException
Thrown to indicate that a method has been passed an illegal or inappropriate argument.
```

Reply:
okay yea makes sense!